### ISSUE TITLE
Performance issue with arrays

### ISSUE THREAD

**Issue by Nafees10** [Nafees10/qscript#10]
This code:
```
int[] (a);
int (i);
i = 0;
while ( i < 100000 ){
    a = a ~ array (i);
    i = i + 1;
}
```
Is slow, as expected. But this code:
```
int[] (a);
int (i);
a = setLength(a, 100000);
i = 0;
while (i < 100000){
    a[i] = i;
    i = i + 1;
}
```
Is surprisingly slower than the first one!
My guess is that this is because concatenating to array is less instructions in the bytecode than setting the value of an element.

**Comment by Nafees10** [comment]
I plan to fix this is the coming version, by adding a new instruction to modify an element in array, so it doesn't have to be copied to stack, modified, and set to var's value. All this should be done by single instruction

**Comment by Nafees10** [comment]
Partially Fixed by merge #16 
No new instruction added, but other performance improvements help the attached code run way faster than in 0.6.x
